src/ok.rs

```diff
@@ -605,3 +671,209 @@ fn memlock_rlimit() -> TestResult {
+        Ok(raw) => raw,
+        Err(e) => return TestResult{
+                    name: "Secure Nested Paging (SNP)".to_string(),
+                    stat: TestState::Fail,
+                    mesg: format!("MSR read failed: {}", e).into()
+        }
+    };
+    let (testres,mesg) = match raw_value >> 24 & 1{
+        1 => (TestState::Pass, "Enabled"),
+        0 => (TestState::Fail, "Disabled"),
+        _ => (TestState::Fail, "Invalid Number Found")
+    };
+    TestResult { 
+        name: "Secure Nested Paging (SNP)".to_string(), 
+        stat: testres, 
+        mesg: Some(mesg.to_string())
+    }
+}
+
+fn get_rmp_address() -> TestResult {
+    let rmp_base = match msr_bit_read(0xC0010132, 0){
+        Ok(raw) => raw,
+        Err(e) => return TestResult{
+            name: "Reading RMP table".to_string(),
+            stat: TestState::Fail,
+            mesg: format!("Failed to read the desired MSR: {}",e).into()
+        }
+    };
+    let rmp_end = match msr_bit_read(0xC0010133, 0){
+        Ok(raw) => raw,
+        Err(e) => return TestResult{
+            name: "Reading RMP table".to_string(),
+            stat: TestState::Fail,
+            mesg: format!("Failed to read the desired MSR: {}",e).into()
+        }
+    };
+    if rmp_base == 0 || rmp_end == 0 {
+        TestResult{
+            name: "Reading RMP table".to_string(),
+            stat: TestState::Fail,
+            mesg: format!("RMP table was not read successfully").into()
+        }
+    } else{
+        TestResult{
+            name: "Finding RMP table addresses".to_string(),
+            stat: TestState::Pass,
+            mesg: format!("RMP table was read, addresses: {} - {}",rmp_base,rmp_end).into()
+        }    
+    }
+
+}
+
+fn snp_ioctl(test: &str) -> TestResult {
+    let status = match snp_platform_status() {
+        Ok(stat) => stat,
+        Err(e) => return TestResult {
+            name: "Looking at SNP platform status".to_string(),
+            stat: TestState::Fail,
+            mesg: format!("Error finding SNP platform status: {}",e).into()
+        }
+    };
+
+    if test == "tcb" {  
+        if status.platform_tcb_version == status.reported_tcb_version {
+            TestResult{
+                name: "Comparing TCB values".to_string(),
+                stat: TestState::Pass,
+                mesg: format!("TCB versions match \n\n Platform TCB version: {} \n Reported TCB version: {}", 
+                            status.platform_tcb_version, status.reported_tcb_version).into()
+            }
+        }
+        else {
+            TestResult{
+                name: "Comparing TCB values".to_string(),
+                stat: TestState::Fail,
+                mesg: format!("The TCB versions did NOT match").into()
+            }
+        }
+    }
+    else if test == "rmp init"{
+        if status.is_rmp_init == 1 {
+            TestResult{
+                name: "Checking if RMP init".to_string(),
```

Comment:
```suggestion
                name: "RMP INIT".to_string(),
```
I just like how this reads better